## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [16]:
import pandas as pd
import numpy as np


Defaulting to user installation because normal site-packages is not writeable


In [17]:
import os
os.getcwd()

'C:\\Users\\Юлія Матіюк\\Python for DA_homework'

In [18]:
os.listdir ("../Data for homework/")

['ab_data.csv',
 'cookie_cats.csv',
 'data.csv',
 'data_2.csv',
 'medical-charges.csv',
 'supermarket_sales.csv',
 'yulu_rental.csv']

In [19]:
data_path = "../Data for homework/data.csv"
df= pd.read_csv (data_path)

In [20]:
data = df['data'].to_numpy()

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [21]:
print(data)

[17.49945292 19.66239904  7.18282343 29.84162485  9.23938649 14.94951581
 23.0172885  12.52827148 13.65228669 14.54595431 23.30872427 33.75324808
 20.24923636 13.29244733 23.23434992 16.4230418  19.88521702 27.05000732
 15.5127743  20.05415151 14.73135264 19.06139498 21.53942271 14.06732571
 17.9670682  18.58289581 16.17406993 12.87432628 11.47269664 19.07902883
 18.38565824 33.38820073  5.39139454 20.67635903 22.22266722 28.15780318
 23.01114324 14.93471778 20.00005857 23.25411543 18.11895082 24.62607043
  8.79145607 30.387108   28.80606806 17.98593597 23.66804468 20.28782355
 15.02518827 20.52626131 26.00219532 17.71344489 17.74598346 19.55317542
 22.60097798 27.67027538 16.19192417 23.05037746 21.29669604  8.84832568
 17.48410111 19.20602661 19.76257856 21.9560206   7.75806171 20.27753314
 15.93394654 11.36336584 23.14577858 24.41167746 16.08049839 25.05473769
 17.71090111 20.39893405 13.40756632 29.50692234  4.04330326 19.45128426
 24.17071763  7.79920072 18.86318441 19.53668801 24

In [22]:
data.shape

(100,)

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

In [24]:
!pip install statsmodels

Defaulting to user installation because normal site-packages is not writeable


In [26]:
from statsmodels.stats.weightstats import ztest

In [28]:
print('mean=%.2f stdv=%.2f' % (np.mean(data), np.std(data)))

mean=19.38 stdv=6.22


In [30]:
data

array([17.49945292, 19.66239904,  7.18282343, 29.84162485,  9.23938649,
       14.94951581, 23.0172885 , 12.52827148, 13.65228669, 14.54595431,
       23.30872427, 33.75324808, 20.24923636, 13.29244733, 23.23434992,
       16.4230418 , 19.88521702, 27.05000732, 15.5127743 , 20.05415151,
       14.73135264, 19.06139498, 21.53942271, 14.06732571, 17.9670682 ,
       18.58289581, 16.17406993, 12.87432628, 11.47269664, 19.07902883,
       18.38565824, 33.38820073,  5.39139454, 20.67635903, 22.22266722,
       28.15780318, 23.01114324, 14.93471778, 20.00005857, 23.25411543,
       18.11895082, 24.62607043,  8.79145607, 30.387108  , 28.80606806,
       17.98593597, 23.66804468, 20.28782355, 15.02518827, 20.52626131,
       26.00219532, 17.71344489, 17.74598346, 19.55317542, 22.60097798,
       27.67027538, 16.19192417, 23.05037746, 21.29669604,  8.84832568,
       17.48410111, 19.20602661, 19.76257856, 21.9560206 ,  7.75806171,
       20.27753314, 15.93394654, 11.36336584, 23.14577858, 24.41

In [35]:
null_mean = 20
alpha = 0.05

In [36]:
ztest_Score, p_value = ztest(data, value = null_mean, alternative='larger')

In [37]:
ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

In [38]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхилена.")

Н0 не може бути відхилена.


Для прийняття рішення про статистичну значущість різниці у розмірі середнього чеку, ми порівнюємо p-значення з рівнем значущості альфа.
p-значення = 0.8402164165464012, що значно більше за 0.05.
Таким чином, ми не маємо статистично значущої різниці у розмірі середнього чеку. 
Це означає, що немає достатніх підстав стверджувати, що середній чек за останній місяць відрізняється 
від попереднього середнього значення $20.


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [41]:
os.listdir ("../Data for homework/")

['ab_data.csv',
 'cookie_cats.csv',
 'data.csv',
 'data_2.csv',
 'medical-charges.csv',
 'supermarket_sales.csv',
 'yulu_rental.csv']

In [42]:
data_path = "../Data for homework/data_2.csv"
df= pd.read_csv (data_path)

In [43]:
data_august = df['data'].to_numpy()

In [44]:
print (data_august)

[ 5.94015879 29.18934391 11.38350244 15.43409823 19.89675209 26.35462741
 21.71541042 30.07928738 18.559706    8.98262482 21.75095478 21.99252789
 16.37737276 18.36827303 15.96373794 30.98838431 16.13256609 11.01409438
 21.96585507  9.46690988 23.42563685 21.53704461 13.26051457 17.14026368
 15.42554536 30.6793921  21.26136766 12.18702858 10.69691295 21.57137244
 19.44915472 15.29409103 21.20725805 18.5615031  27.27465429 22.71259245
 19.41187696 21.98647106 13.33332309 26.42199496 23.87611696 16.09133552
 20.95972522 15.47445472 32.79323825 16.42470817 23.61838166 22.92533987
 14.82328184 19.14936908 17.61708292 20.76144078 20.88864519 22.17749704
 24.44099555 17.42064342 19.01311241 17.11869915 14.66455092  9.59578906
 23.93038503 18.80003123 16.62880551  9.30221669 10.74561613  5.00093505
 18.64262674 23.06962246 23.27815123 19.56141603 14.07482487 19.67979235
 20.42825975 23.21152701 25.38472731 18.32183161 18.40471839 23.90049615
 22.82595361 17.8883959   7.69030253 15.65278307 25

In [47]:
os.listdir ("../Data for homework/")

['ab_data.csv',
 'cookie_cats.csv',
 'data.csv',
 'data_2.csv',
 'medical-charges.csv',
 'supermarket_sales.csv',
 'yulu_rental.csv']

In [48]:
data_path = "../Data for homework/data.csv"
df= pd.read_csv (data_path)

In [49]:
data_july = df['data'].to_numpy()

In [50]:
print (data_july)

[17.49945292 19.66239904  7.18282343 29.84162485  9.23938649 14.94951581
 23.0172885  12.52827148 13.65228669 14.54595431 23.30872427 33.75324808
 20.24923636 13.29244733 23.23434992 16.4230418  19.88521702 27.05000732
 15.5127743  20.05415151 14.73135264 19.06139498 21.53942271 14.06732571
 17.9670682  18.58289581 16.17406993 12.87432628 11.47269664 19.07902883
 18.38565824 33.38820073  5.39139454 20.67635903 22.22266722 28.15780318
 23.01114324 14.93471778 20.00005857 23.25411543 18.11895082 24.62607043
  8.79145607 30.387108   28.80606806 17.98593597 23.66804468 20.28782355
 15.02518827 20.52626131 26.00219532 17.71344489 17.74598346 19.55317542
 22.60097798 27.67027538 16.19192417 23.05037746 21.29669604  8.84832568
 17.48410111 19.20602661 19.76257856 21.9560206   7.75806171 20.27753314
 15.93394654 11.36336584 23.14577858 24.41167746 16.08049839 25.05473769
 17.71090111 20.39893405 13.40756632 29.50692234  4.04330326 19.45128426
 24.17071763  7.79920072 18.86318441 19.53668801 24

In [51]:
np.mean (data_july), np.mean (data_august)

(19.37755319670441, 19.37322276512519)

In [52]:
ztest(data_july, data_august, value = 0)

(0.005246673456808504, 0.9958137794593356)

Висновок: p-значення = 0.9958137794593356, що значно більше за 0.05. Таким чином, ми не маємо статистично значущої різниці у розмірі середнього чеку між липнем та серпнем 2022 року, це також видно з середніх показників чеків в ці місяці. Це означає, що на основі зібраних даних ми не можемо стверджувати, що середній чек у серпні значущо відрізняється від середнього чеку у липні. Нульова гіпотеза не може бути відхилена.